dashboard/test/ui/runner.rb

```diff
@@ -25,6 +26,35 @@
 
 ENV['BUILD'] = `git rev-parse --short HEAD`
 
+S3_LOGS_BUCKET = 'cucumber-logs'
+S3_LOGS_PREFIX = rack_env?(:test) ? 'test' : 'development'
+
+#
+# Upload the given filename (of a cucumber log) to the logs s3 bucket.
+# Returns a public URL to the uploaded log.
+#
+def upload_log_to_s3(filename, s3_client)
```

Comment:
Great idea, & support extracting this logic from runner.rb. Worth noting there is a utility `S3.upload_to_bucket` https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/aws/s3.rb#L43-L43 which grabs a new S3 client and uploads a file (but does not return the version #, so might not be perfect for this use case). That might be a good place to throw such a utility.